bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -318,29 +318,30 @@ public boolean removeFutures(Runnable task) {
             ArrayList<Future<?>> obsoleteFutures = new ArrayList<Future<?>>();
             synchronized (futures) {
                 ArrayList<Future<?>> taskFutures = futures.get(task);
-                if (taskFutures.size() != 0) {
-                    logger.trace("Runnable '{}' has {} Futures to be removed", task.toString(), taskFutures.size());
-                    for (Future<?> future : taskFutures) {
-                        future.cancel(false);
-                        timestamps.remove(future);
-                        obsoleteFutures.add(future);
+                if (taskFutures != null) {
+                    if (taskFutures.size() != 0) {
+                        logger.trace("Runnable '{}' has {} Futures to be removed", task.toString(), taskFutures.size());
```

Comment:
You should not call toString yourself. This is done by the logger if needed (now toString is called all the time regardless if log level is trace).
This should be changed.